coral-common/src/main/java/com/linkedin/coral/common/functions/FunctionReturnTypes.java

```diff
@@ -56,9 +56,15 @@ private FunctionReturnTypes() {
   public static final SqlReturnTypeInference ARRAY_OF_ARG0_TYPE =
       opBinding -> opBinding.getTypeFactory().createArrayType(opBinding.getOperandType(0), -1);
 
-  public static SqlReturnTypeInference arrayOfType(final SqlTypeName typeName) {
-    return opBinding -> opBinding.getTypeFactory().createArrayType(opBinding.getTypeFactory().createSqlType(typeName),
-        -1);
+  public static SqlReturnTypeInference arrayOfType(final SqlTypeName typeName, boolean elementsNullable) {
+    return opBinding -> {
+      RelDataTypeFactory typeFactory = opBinding.getTypeFactory();
+      RelDataType relType = typeFactory.createSqlType(typeName);
+      RelDataType arrayType = typeFactory.createArrayType(relType, -1);
+      if (!elementsNullable)
+        return arrayType;
+      return typeFactory.createTypeWithNullability(arrayType, true);
+    };
```

Comment:
If the original `jobentitlementids` is like `array<bigint>`, I think this logic will make it to be `[array<bigint>, null]` rather than `array<[bigint, null]>` (`[x, null]` means `x` is nullable)
I am not sure if that is what we want, if it is, I am fine with this change.

Reply:
Thanks for pointing out. Modified the method and validated that read from affected view goes through.